addons/account/migrations/11.0.1.1/post-migration.py

```diff
@@ -91,21 +91,47 @@ def fill_account_invoice_line_total(env):
     rest_lines = line_obj.search([]) - empty_lines - simple_lines
     openupgrade.logger.debug("Compute the rest of the account.invoice.line"
                              "totals: %s" % len(rest_lines))
-    for line in rest_lines:
-        # avoid error on taxes with other type of computation ('code' for
-        # example, provided by module `account_tax_python`). We will need to
-        # add the computation on the corresponding module post-migration.
-        types = ['percent', 'fixed', 'group', 'division']
-        if any(x.amount_type not in types for x in line.invoice_line_tax_ids):
-            continue
-        # This has been extracted from `_compute_price` method
-        currency = line.invoice_id and line.invoice_id.currency_id or None
-        price = line.price_unit * (1 - (line.discount or 0.0) / 100.0)
-        taxes = line.invoice_line_tax_ids.compute_all(
-            price, currency, line.quantity, product=line.product_id,
-            partner=line.invoice_id.partner_id,
+
+    openupgrade.logged_query(
```

Comment:
Why doing all of this? Isn't enough to write `price_total` by SQL with previous code?

Reply:
The problem here was the for loop over 6 million invoice lines that was taking too long when in fact a grouping of those lines can be done. If you find lines that share the same invoice_line_tax_ids, price, currency, quantity and product, for all those invoice lines the price_total will be the same. That is the reason why the first SQL query to group and then the second to compute the price_total for every group. The for loop now is way smaller than before.

Reply:
But that's very unlikely to happen. If the problem is the loop over a great number, then simply use chunk method in openupgradelib for splitting it. I consider this code very ugly and hard to debug/analyze. And a select never uses `loggued_query`!

Please try what I suggest to see the performance.

Reply:
I do not think what you propose would have any effect at all. I don't see the difference between iterating once over 6 million records than 100 times over 60000 records. Overall time remains the same IMO but maybe I'm missing something here.
Regarding the code I know that is not very clean, I have added comments to make it more understandable.

Reply:
Why don't you try with both suggestions (chunk + update by SQL)? Please have faith :angel:

Reply:
Yes, I am going to try

Reply:
Wait, I like this approach! I happen to be working on a migration in which this can drastically lower the duration as well due to the uniformity of the invoice lines.

Reply:
Let's see any way the numbers with what I have been talking.